projects/shared/src/lib/components/building/building.component.ts

```diff
@@ -1,27 +1,89 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ViewChild, ElementRef, AfterViewInit } from '@angular/core';
 import { ActivatedRoute, Router } from '@angular/router';
 import { FloorInfo } from '../../models/floor-info';
+import {HostListener} from '@angular/core';
 
 @Component({
   selector: 'shared-building',
   templateUrl: './building.component.html',
   styleUrls: ['./building.component.scss']
 })
-export class BuildingComponent implements OnInit {
-  public maxWidth: string;
+export class BuildingComponent implements AfterViewInit {
+
+  @ViewChild('floorScroll') floorScroll: ElementRef;
+  public floorsInfo: FloorInfo[];
+  public isScrollOnBottom: boolean;
+  public isScrollOnTop: boolean;
+
   constructor(private route: ActivatedRoute, private router: Router) {
     this.route.data.subscribe(data => {
       this.floorsInfo = data.floorsInfo;
       this.floorsInfo.reverse();
     });
   }
 
-  public floorsInfo: FloorInfo[];
-  ngOnInit() {
-    const width = ((document.documentElement.clientHeight - 140) / (this.floorsInfo.length + 3)) * 6;
-    this.maxWidth = width + 'px';
+  ngAfterViewInit() {
+    const floorsCount = this.floorsInfo.length;
+    const floorsOverScroll = floorsCount - 5;
+    this.floorScroll.nativeElement.onscroll = () => {
+      if (this.floorScroll.nativeElement.scrollTop >
+        ((this.floorScroll.nativeElement.scrollHeight / floorsCount) * floorsOverScroll - 5)) {
+        this.isScrollOnBottom = true;
```

Comment:
this.isScrollOnBottom = this.floorScroll.nativeElement.scrollTop >
        ((this.floorScroll.nativeElement.scrollHeight / floorsCount) * floorsOverScroll - 5)